File: src/main/java/net/lumalyte/trivia/models/PlayerStats.java
package net.lumalyte.trivia.models;

import java.util.UUID;

public class PlayerStats implements Comparable<PlayerStats> {
    private final UUID playerId;
    private final String playerName;
    private int totalCorrect;
    private int easyCorrect;
    private int mediumCorrect;
    private int hardCorrect;
    private int points;

    public PlayerStats(UUID playerId, String playerName) {
        this.playerId = playerId;
        this.playerName = playerName;
        this.totalCorrect = 0;
        this.easyCorrect = 0;
        this.mediumCorrect = 0;
        this.hardCorrect = 0;
        this.points = 0;
    }

    public void addCorrectAnswer(String difficulty) {
        totalCorrect++;
        switch (difficulty.toLowerCase()) {
            case "easy":
                easyCorrect++;
                points += 1;
                break;
            case "medium":
                mediumCorrect++;
                points += 2;
                break;
            case "hard":
                hardCorrect++;
                points += 3;
                break;
        }
    }

    public UUID getPlayerId() {
        return playerId;
    }

    public String getPlayerName() {
        return playerName;
    }

    public int getTotalCorrect() {
        return totalCorrect;
    }

    public int getEasyCorrect() {
        return easyCorrect;
    }

    public int getMediumCorrect() {
        return mediumCorrect;
    }

    public int getHardCorrect() {
        return hardCorrect;
    }

    public int getPoints() {
        return points;
    }

    @Override
    public int compareTo(PlayerStats other) {
        return Integer.compare(other.points, this.points); // Sort by points descending
    }
} 

File: src/main/kotlin/net/badgersmc/trivia/domain/PlayerStats.kt
package net.badgersmc.trivia.domain

import java.util.UUID

/** Per-player trivia statistics. Sorts by points descending. */
data class PlayerStats(
val playerId: UUID,
var playerName: String,
    var totalCorrect: Int = 0,
    var easyCorrect: Int = 0,
    var mediumCorrect: Int = 0,
    var hardCorrect: Int = 0,
    var points: Int = 0,
) : Comparable<PlayerStats> {

    /** Increments counters based on difficulty using configurable point values. */
    fun addCorrectAnswer(difficulty: String, easyPoints: Int = 1, mediumPoints: Int = 2, hardPoints: Int = 3) {
        totalCorrect++
        when (difficulty.lowercase()) {
            "easy" -> { easyCorrect++; points += easyPoints }
            "medium" -> { mediumCorrect++; points += mediumPoints }
            "hard" -> { hardCorrect++; points += hardPoints }
        }
    }

    /** Sorts descending by points (higher points = "less than" in comparison). */
    override fun compareTo(other: PlayerStats): Int =
        other.points.compareTo(points)
}
